purchase_order_update_planned_date/models/purchase_order.py

```diff
@@ -0,0 +1,17 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See https://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models
+from datetime import timedelta
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    def button_confirm(self):
+        super().button_confirm()
+        for order in self:
+            if order.date_order and order.date_planned:
+                dif = (order.date_approve - order.date_order).days
+                new_date_planned = order.date_planned + timedelta(days=dif)
+                order.write({'date_planned': new_date_planned})
+                for picking in order.picking_ids:
+                    picking.write({'scheduled_date':new_date_planned})
```

Comment:
Is this really needed? In other words, if this code is removed, pickings `scheduled_date` are not updated?

Reply:
no, scheduled_date takes the value of planned_date before updating it

Reply:
OK, could you investigate in `purchase_stock` addon in which situations `scheduled_date` is originally set depending on `planned_date`?

Reply:
In `purchase_stock` there is nothing to relate these fields

Reply:
Function `_prepare_stock_move_vals` returns values to create a stock.move record. Among them is `date` (which would be `scheduled_date`) which takes the value of `date_planned`.
https://github.com/odoo/odoo/blob/15.0/addons/purchase_stock/models/purchase.py#L533

Reply:
Take a look a this:

https://github.com/odoo/odoo/blob/3a28e5b0adbb36bdb1155a6854cdfbe4e7f9b187/addons/stock/models/stock_picking.py#L549

Picking `scheduled_date` is based on moves `date` field. And date for moves actually depends on order `planned_date`. So if we update `date_planned` _before_ the order is confirmed (in this implementation we make this _after_ confirmation), this could be enough (when confirm operation is made, `date_planned` is already the desired one at this moment, so moves are created at the first time with the right date and we don't need to update `schedule_date` for picking, because is actually updated at this point). Is that possible or is it complicated? I don't really know